packages/discovery-react-components/src/components/CIDocument/utils/__tests__/filterUtils.spec.ts

```diff
@@ -56,14 +63,17 @@ describe('filterUtils', () => {
       filter = updateFilter(filter, filterPartA2);
       filter = updateFilter(filter, filterPartB1);
       filter = updateFilter(filter, filterPartB2);
+      filter = updateFilter(filter, filterPartC1);
 
       expect(filter.FILTER_GROUP_A.length).toBe(2);
       expect(filter.FILTER_GROUP_B.length).toBe(1);
-      expect(filter.FILTER_GROUP_C).toBe(undefined);
+      expect(filter.FILTER_GROUP_C.length).toBe(0);
+      expect(filter.FILTER_GROUP_D).toBe(undefined);
```

Comment:
You can directly check for `.toBeUndefined()` here and in other place down, which is recommended over referring to `undefined` directly.